From 9425ddcc3d9ee90a963732863ff76bee0496e54b Mon Sep 17 00:00:00 2001
From: shandongtlb <40312509+shandongtlb@users.noreply.github.com>
Date: Sat, 19 Sep 2026 08:26:50 +0000
Subject: [PATCH 18/22] certs: preserve stock GKI public certificate trust

Embed the public certificate extracted from the target stock Image, without adding local signing trust or changing signature checks. Keep the local signing key/certificate pair for official build goals; local generated modules are not stock-signed replacements. Stock DER SHA256: 9be2c27bf7671d3935bfa2379bbd534d7ca3f46a31531b3db80f52aa051775d3.
---
 certs/Makefile              |  7 +++++++
 certs/oneplus13-stock.pem   | 31 +++++++++++++++++++++++++++++++
 certs/system_certificates.S |  2 +-
 3 files changed, 39 insertions(+), 1 deletion(-)
 create mode 100644 certs/oneplus13-stock.pem

diff --git a/certs/Makefile b/certs/Makefile
index 799ad7b9e..19af70e3d 100644
--- a/certs/Makefile
+++ b/certs/Makefile
@@ -27,6 +27,13 @@ quiet_cmd_extract_certs  = CERT    $@
 extract-cert-in = $(filter-out $(obj)/extract-cert, $(real-prereqs))
 
 $(obj)/system_certificates.o: $(obj)/x509_certificate_list
+$(obj)/system_certificates.o: $(obj)/stock_gki_certificate.x509
+
+# Keep the stock Image trust anchor separate from the local module signing key.
+$(obj)/stock_gki_certificate.x509: $(srctree)/$(src)/oneplus13-stock.pem $(obj)/extract-cert FORCE
+	$(call if_changed,extract_certs)
+
+targets += stock_gki_certificate.x509
 
 $(obj)/x509_certificate_list: $(CONFIG_SYSTEM_TRUSTED_KEYS) $(obj)/extract-cert FORCE
 	$(call if_changed,extract_certs)
diff --git a/certs/oneplus13-stock.pem b/certs/oneplus13-stock.pem
new file mode 100644
index 000000000..0cdb6c1b5
--- /dev/null
+++ b/certs/oneplus13-stock.pem
@@ -0,0 +1,31 @@
+-----BEGIN CERTIFICATE-----
+MIIFSTCCAzGgAwIBAgIUJnKUPNnLiXJpq7C8aSskhl4Hn4MwDQYJKoZIhvcNAQEF
+BQAwLjEsMCoGA1UEAwwjQnVpbGQgdGltZSBhdXRvZ2VuZXJhdGVkIGtlcm5lbCBr
+ZXkwIBcNMjYwNDA5MDEzNzUyWhgPMjEyNjAzMTYwMTM3NTJaMC4xLDAqBgNVBAMM
+I0J1aWxkIHRpbWUgYXV0b2dlbmVyYXRlZCBrZXJuZWwga2V5MIICIjANBgkqhkiG
+9w0BAQEFAAOCAg8AMIICCgKCAgEAwWV+zIxlR301+i9QLoJFql5z+6kv9X53wcVi
+t8RmOObmCvWqzJK6QfQPqBsbRgCj5jKdk3sVz680nlKDlhJexxn38ipyay9069VK
+VKUFxBNgosBHQrYsDbZ7q1L9Z/WDH+RBfks+6gNATpCLHurfg1mkuntvBXryHq8K
+OylBB52A/8Ni4lKZtKjNY3eSILqmAXBI4rNTaOHbcqml1mp4a1JzoFta+Tor6flc
+IZu2Xj5BieezXtOmGYcKrQCbRi0Si6ddq7RHpo+K6duymVpkMKuIG59Q3/XxroZ/
+xZ5kz6bDQw1w4PZ6NQ9q3B+LAhtkSyJZQmxFsA+EWNAcni9HWYgn3UgGSOlyHzE5
+/oepxKLfUvLqG9VQ1oN/gVGy8HzRyFQ+/bSUGVVHal5dRrhCMuW1drQKQ/Qb3TPn
+N2GI0etb0DzfJrRrYKHUX3jRubZL8Wj+KUQiBnhIZmbhNhYTZ/cU8gvuw5slmTDv
+5tYkg1rq71vTYmD5WxSX1u2KRaTcOWd+H/rLekBBT1fvLVw7TKE9NnBAmt+l5Zb6
+GDF8gmhV1huXJ+8ot6FFG9g4jQCmVCek/mjuq/NgdCOX9awmF35iQEOg9gGN9MNg
+ca2OV29KdUo14m8OTx5bwk/DpoyKKU7/MnSftyn5JP16528qN0oEplp6xSPJSoX/
+bKPcAskCAwEAAaNdMFswDAYDVR0TAQH/BAIwADALBgNVHQ8EBAMCB4AwHQYDVR0O
+BBYEFO5iPDypZk8HFoxZJ8xCLntTbmkHMB8GA1UdIwQYMBaAFO5iPDypZk8HFoxZ
+J8xCLntTbmkHMA0GCSqGSIb3DQEBBQUAA4ICAQCSOVfAN8kXvfQ7fOicmZLwaZuQ
+DEn5NdWoXSkq6H/LUjoxIt62IabjusdIsSrThA1aBO5oRNWZRC/iHnafr8iJ5qyq
+x6rOSjgCzBAJKPhk07UhapsHa1U7cTY2Ztrp1GEARLbCvRbG2x+nKEPno9WjaMGl
+dPXgHFcOnE2860iukVtO7ELQ6oTYx2q18ezm9FF6F74dIAqqQIe3ksB4IIdffkDs
+wkS0RNjVC/rKxsrS/+LHRMbDf0rFkBDVLT9OXI141/M+C4+DE0Hqg+9jf7dyyGCo
+gYYSupBAoYvVOr0lz45Yz13qlutkYC71RlgOPzWx9lkOUWhg7v/68GoxENbi0Dyp
+H3WY3TguzyQ83MMfxKsu0198mzqr/QL62WK/izQrfgyxkb3GaEZoHTD8onZv1tp0
+jaaBEV8M5qGvnU1cAmea1bpJrL8EadMcI10KFNsUm2e0O/Wml6FIc6HjOTryYUAR
+Hd7Nyeu6ohCeyK/hh8PToPVUTBbaP5zmauhMs5bwj0FA4+0Mb2bpTOe3BKOU/3OD
+oeYK7YXb/QNG6XWUww10aBfzldA1BnbMsogeEKse0tr6yrsP4RaWX3RmH+Y/3bRy
+74cGZVIkzW2ReaqB4uwSO0ymxu2BgA/mQGrBwZimA/Yneoe6PzJ5Nq899ORENzSi
+SUOP9yBFR+1uWuzCEg==
+-----END CERTIFICATE-----
diff --git a/certs/system_certificates.S b/certs/system_certificates.S
index 003e25d4a..ba87856e5 100644
--- a/certs/system_certificates.S
+++ b/certs/system_certificates.S
@@ -9,7 +9,7 @@
 system_certificate_list:
 __cert_list_start:
 __module_cert_start:
-	.incbin "certs/signing_key.x509"
+	.incbin "certs/stock_gki_certificate.x509"
 __module_cert_end:
 	.incbin "certs/x509_certificate_list"
 __cert_list_end:
